subprojects/errorprone/src/main/java/org/mockito/errorprone/bugpatterns/MockitoMockedStatic.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2017 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.errorprone.bugpatterns;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.VariableTreeMatcher;
+import com.google.errorprone.matchers.Description;
+import com.sun.source.tree.VariableTree;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "MockitoMockedStatic",
+        summary = "Fields or parameters annotated with @Mock in MockedStatic can lead to compilation issues.",
+        explanation = "Fields or parameters annotated with @Mock in MockedStatic can lead to compilation issues.",
+        severity = SeverityLevel.ERROR)
+public class MockitoMockedStatic extends BugChecker implements VariableTreeMatcher {
+
+  @Override
+  public Description matchVariable(VariableTree tree, VisitorState state) {
+    if (tree.getModifiers().getAnnotations().stream()
+            .anyMatch(annotation ->
+                    state.getSourceForNode(annotation).equals("@org.mockito.Mock")
```

Comment:
Do not use `getSourceForNode` to compare annotations, as that is equivalent to comparing `toString()`. Instead, use https://errorprone.info/api/latest/com/google/errorprone/util/ASTHelpers.html#isSameType(com.sun.tools.javac.code.Type,com.sun.tools.javac.code.Type,com.google.errorprone.VisitorState)